types/braintree/index.d.ts

```diff
@@ -978,6 +976,16 @@ declare namespace braintree {
         trialPeriod?: boolean;
     }
 
+    export interface SubscriptionUpdateRequest extends SubscriptionRequest {
```

Comment:
From what I'm seeing on the docs, the `id` and `merchantAccountId` as part of the update request object seem to be optional as well. I think we can keep the existing `SubscriptionRequest` interface and make both these fields optional on that interface.

Reply:
But then we will have a slightly incorrect check for both update and create. Wouldn't that be a problem?

Reply:
Not sure I follow, what do you mean by slightly incorrect check? The `id` is required for update but only as the first argument in the `update` method, not as part of the request object itself. The way I'm reading the docs, the request object marks both of the touched fields as optional.

Reply:
Actually it seems that `id` is optional in `create` but required in `update`
While `merchantAccountId` is optional in both of them

Reply:
@sharno So, there are two separate `id`s we may be talking about here. There is an `id` passed as the first argument to the `update` method. This is required, and is currently marked as such in the TS definition.

The second `id` is passed as part of the request object, and per the docs says "If this option is provided", leading me to believe it is optional.

Note the differences between:
1) The id argument (https://developers.braintreepayments.com/reference/request/subscription/update/node#arg.id)
2) The id as part of the request object (https://developers.braintreepayments.com/reference/request/subscription/update/node#id)

Reply:
@smrubin Oh I see what you are referring to now. Your suggestion makes sense then of just making both fields optional in the current TS definition.